### PR TITLE
ci: enable clippy 2018 idiom check

### PR DESCRIPTION
**Pull request by Mirko-von-Leipzig**
Adds `--deny rust_2018_idioms` to CI and fixes the existing code issues.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -41,7 +41,7 @@ jobs:
         with:
           components: clippy
       - uses: Swatinem/rust-cache@v1
-      - run: cargo clippy --workspace --all-targets -- -D warnings
+      - run: cargo clippy --workspace --all-targets -- -D warnings -D rust_2018_idioms
```
Comment (koivunej) [inline .github/workflows/ci.yml:44]: I would put this at the `crates/pathfinder/src/lib.rs` so there'd be more instant feedback.

**Review by koivunej — APPROVED** [review]
Approve with the move from clippy command line to crates/pathfinder/src/lib.rs ... Aah yes, I guess it needs to be put into every lib.rs to have it workspace wide.

**Comment by Mirko-von-Leipzig** [comment]
> Approve with the move from clippy command line to crates/pathfinder/src/lib.rs ... Aah yes, I guess it needs to be put into every lib.rs to have it workspace wide.

Yeah its a pity one can't specify a clippy file for the crate.

**Comment by koivunej** [comment]
You can remove it from the workflow cmdline.

**Pull request merged**